['text':'*
 *    Copyright (C) 2019-present MongoDB, Inc.
 *
 *    This program is free software: you can redistribute it and/or modify
 *    it under the terms of the Server Side Public License, version 1,
 *    as published by MongoDB, Inc.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    Server Side Public License for more details.
 *
 *    You should have received a copy of the Server Side Public License
 *    along with this program. If not, see
 *    <http://www.mongodb.com/licensing/server-side-public-license>.
 *
 *    As a special exception, the copyright holders give permission to link the
 *    code of portions of this program with the OpenSSL library under certain
 *    conditions as described in each individual source file and distribute
 *    linked combinations including the program with the OpenSSL library. You
 *    must comply with the Server Side Public License in all respects for
 *    all of the code used other than as permitted herein. If you modify file(s)
 *    with this exception, you may extend this exception to your version of the
 *    file(s), but you are not obligated to do so. If you do not wish to do so,
 *    delete this exception statement from your version. If you delete this
 *    exception statement from all source files in the program, then also delete
 *    it in the license file.
 ','line_number':1,'multiline':True]
['text':' namespace','line_number':53,'multiline':False]
['text':' mustExit is set when the clone has been canceled externally.','line_number':165,'multiline':False]
['text':' If lastError is set, this is a retry.','line_number':169,'multiline':False]
['text':' Execute any per-retry logic needed by the cloner.','line_number':189,'multiline':False]
['text':' The promise has to be a class variable to correctly return the error code in the case','line_number':223,'multiline':False]
['text':' where executor->scheduleWork fails (i.e. when shutting down)','line_number':224,'multiline':False]
['text':' The _promise can run the error callback on this thread, so we must not hold the lock','line_number':232,'multiline':False]
['text':' when we set it.','line_number':233,'multiline':False]
['text':' namespace repl','line_number':295,'multiline':False]
['text':' namespace mongo','line_number':296,'multiline':False]
